### PR TITLE
Adds server based on latest master

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
This image pulls always latest master and install it at /nc.
Also a test user with "test" password is created.

THis is needed for automated testing on android, as it needs a server to do basic operations on it.

All credits belong to @rullzer

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Yesterday I noticed that this one had some bugs ... for example are thumbnails broken and not generated. So I migrated my local docker image to be Debian based and have less selfcompiled stuff, but relies more on the real world instances (a Debian with standard packages that are out there).

`Dockerfile`:

```
FROM debian:stretch

RUN apt-get update \
    && apt-get install -y \
       php \
       php-curl \
       php-gd \
       php-imagick \
       php-mbstring \
       php-pdo-mysql \
       php-pdo-pgsql \
       php-redis \
       php-sqlite3 \
       php-xml \
       php-zip \
    && apt-get autoremove -y \
    && apt-get autoclean \
    && apt-get clean \
    && rm -rf /tmp/* /var/tmp/* /var/lib/apt/lists/*

COPY php.ini /etc/php/7.0/apache2/conf.d/30-php.ini
COPY apache2-foreground /usr/local/bin/

WORKDIR /var/www/html

EXPOSE 80
CMD ["apache2-foreground"]
```

`php.ini` for opcache and better performance

```ini
opcache.enable=1
opcache.enable_cli=1
opcache.interned_strings_buffer=8
opcache.max_accelerated_files=10000
opcache.memory_consumption=128
opcache.save_comments=1
opcache.revalidate_freq=1
```

My entry point script (basically loads the default environment variables of apache and starts it in the foreground):

```
#!/bin/bash
set -e

. /etc/apache2/envvars

apache2 -DFOREGROUND "$@"
```

File: server/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+WORKDIR /var/www/html
+
+RUN git clone https://github.com/nextcloud/server.git nc
+RUN cd nc && git submodule update --init
+RUN chown -R www-data:www-data nc
+RUN chsh -s /bin/bash www-data
+
+ADD init.sh /initnc.sh
+RUN chmod +x /initnc.sh
```
Comment (MorrisJobke) [inline server/Dockerfile]: Apply this to the file and `git commit` this then ;)

File: server/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+       opcache \
+       posix \
+       simplexml \
+       zip \
+    && docker-php-ext-configure gd --with-freetype-dir=/usr/include/ --with-jpeg-dir=/usr/include/ \
+    && docker-php-ext-install -j$(nproc) gd 
+
+WORKDIR /var/www/html
+
+RUN git clone https://github.com/nextcloud/server.git nc
```
Comment (MorrisJobke) [inline server/Dockerfile]: Why are you cloning to `nc` and not to the working dir directly. Then there wouldn't be a need to have it in a subdir like `localhost/nc` ;)

File: server/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+RUN cd nc && git submodule update --init
+RUN chown -R www-data:www-data nc
+RUN chsh -s /bin/bash www-data
```
Comment (MorrisJobke) [inline server/Dockerfile]: Any need for this? You could also simply run the process in the container as root ... there isn't anything other in this container anyways ;)

File: server/init.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+export BRANCH=${BRANCH:=master}
+
+#Where we do all the work
+cd /var/www/html/nc
+
+#Update code
+su www-data -c "
```
Comment (MorrisJobke) [inline server/init.sh]: And also you don't need to do this hack if you run apache as root ;)

File: server/Dockerfile (hunk 3 above)
Comment (tobiasKaminsky) [reply server/Dockerfile]: Roeland said that it was not working if /nc is not owned by apache user

Comment (MorrisJobke) [reply server/Dockerfile]: But if you run apache as root it is owned by it ;) At least that is what works totally fine over here ;)

**Comment by tobiasKaminsky** [comment]
@MorrisJobke I removed now the www-data stuff, but now I get an internal error, when accessing server.

**Comment by tobiasKaminsky** [comment]
@MorrisJobke is seems that it has to be run as www-data.
This is working now, but I doubt that all my changes are correctly ;-)

**Comment by MorrisJobke** [comment]
Looks good so far. Just try it then you know if it works ;)

File: server/init.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+#init
+php occ maintenance:install --admin-user=admin --admin-pass=admin
+OC_PASS=test php occ user:add --password-from-env -- test
+
+#Trusted domains
+php occ config:system:set trusted_domains 1 --value=*
+"
+
+bash apache2-foreground
```
Comment (MorrisJobke) [inline server/init.sh]: really? Just copy paste it in here ;)

**Pull request merged**